AutoCollection/AutoCollector.ts

```diff
@@ -0,0 +1,171 @@
+import AutoCollectConsole = require("./Console");
+import AutoCollectExceptions = require("./Exceptions");
+import AutoCollectPerformance = require("./Performance");
+import AutoCollecPreAggregatedMetrics = require("./PreAggregatedMetrics");
+import HeartBeat = require("./HeartBeat");
+import AutoCollectHttpDependencies = require("./HttpDependencies");
+import AutoCollectHttpRequests = require("./HttpRequests");
+import TelemetryClient = require("../Library/NodeClient");
+import { AutoCollectNativePerformance, IDisabledExtendedMetrics } from "./NativePerformance";
+import { JsonConfig } from "../Library/JsonConfig";
+
+class AutoCollector {
+
+    // Default values
+    public isConsole = true;
+    public isConsoleLog = false;
+    public isExceptions = true;
+    public isPerformance = true;
+    public isPreAggregatedMetrics = true;
+    public isHeartBeat = false;
+    public isRequests = true;
+    public isDependencies = true;
+    public isNativePerformance = true;
+    public isCorrelating = true;
+    public forceClsHooked: boolean;
+    public disabledExtendedMetrics: IDisabledExtendedMetrics;
+
+    private _console: AutoCollectConsole;
+    private _exceptions: AutoCollectExceptions;
+    private _performance: AutoCollectPerformance;
+    private _preAggregatedMetrics: AutoCollecPreAggregatedMetrics;
+    private _heartbeat: HeartBeat;
+    private _nativePerformance: AutoCollectNativePerformance;
+    private _serverRequests: AutoCollectHttpRequests;
+    private _clientRequests: AutoCollectHttpDependencies;
+    private _client: TelemetryClient;
+    private _isStarted = false;
+
+    public setup(client: TelemetryClient) {
+        this._client = client;
+        this._initializeFlagsFromConfig();
+        this._console = new AutoCollectConsole(client);
+        this._exceptions = new AutoCollectExceptions(client);
+        this._performance = new AutoCollectPerformance(client);
+        this._preAggregatedMetrics = new AutoCollecPreAggregatedMetrics(client);
+        this._heartbeat = new HeartBeat(client);
+        this._serverRequests = new AutoCollectHttpRequests(client);
+        this._clientRequests = new AutoCollectHttpDependencies(client);
+        if (!this._nativePerformance) {
+            this._nativePerformance = new AutoCollectNativePerformance(client);
+        }
+    }
+
+    public start() {
+        this._isStarted = true;
+        this._console.enable(this.isConsole, this.isConsoleLog);
```

Comment:
Should the enable methods below only be called if "_isStarted" is false? Otherwise, for this case: 

1. configured flag `false` in json config file
2. call "setxxx" api to config the flag `true`
3. call start

The flag should be `false` because json config always take the highest priority. But in this case, the flag is `true` because setxxx api reset the flag before start. 
A more detailed example: [readme](https://github.com/microsoft/ApplicationInsights-node.js#configure-settings-through-json-file)
If setxxx api takes higher priority, then the `scenario 2` [here](https://github.com/microsoft/ApplicationInsights-node.js#configure-settings-through-json-file) breaks.

Reply:
Flag should be true, explicit API calls in customer code will always have priority over env variables and JSON configuration